spec/other/compatibility_guard_spec.rb

```diff
@@ -4,15 +4,6 @@
   context "when the bundler version is 2+" do
     before { simulate_bundler_version "2.0.a" }
 
-    context "when running on Ruby < 2.3", :ruby => "< 2.3" do
```

Comment:
Any chance to keep this spec, but just "simulate" unsupported Ruby version?

Reply:
The thing is... this spec will never happen in real life I think. Running `gem install bundler` on an unsupported ruby would be short-circuit first by rubygems, wouldn't it?

Reply:
In other words, this situation will only happen if bundler 2 is installed on rubies below 2.3 and that is... impossible I think?

Reply:
I think it is still possible (in theory, not tested) with some old ruby version and old rubygems version not supporting Ruby version constraint. But you're right. It is probably not worth it to keep this spec.

Reply:
Yeah, that's a good point. Maybe it's worth keeping actually.It's just three lines of lib code and this spec.

Reply:
Let's not get too agressive with the removing, I restored the compatibility guard for now :+1:.

Reply:
You can also clone and run Bundler locally, so I think at the bare minimum we should keep the check in lib/